api/server/server.go

```diff
@@ -1123,6 +1126,142 @@ func postContainersCopy(eng *engine.Engine, version version.Version, w http.Resp
 	return nil
 }
 
+// handleArchiveError is a convenience function for the following Copy
+// handlers which makes a JSON error message but leaves writing the
+// header and response to the package's common error handler.
+func handleArchiveError(w http.ResponseWriter, err error) error {
```

Comment:
Functions like `getContainersStatPath()` already are transport aware since they're given an `http.ResponseWriter` and `*http.Request` as arguments.

The jobs they call out to are transport agnostic though and this `handleArchiveError()` simply  sets the response header to be `application/json` and then returns a valid JSON error string. I don't think this conflicts with your changes in #9509, does it?